### ISSUE TITLE
Provide Officially-Supported .NET AOT Images

### ISSUE THREAD

**Issue by simonkurtz-MSFT** [dotnet/dotnet-docker#5020]
We just published container optimizations to the [Azure Container Apps .NET Workshop](https://azure.github.io/aca-dotnet-workshop/aca/12-optimize-containers/#31-table-of-improvements). This workshop provides guidance to our customers using ACA & .NET. Since ACA is Linux only, building for a specific platform with AOT is very suitable and would provide great advantages in terms of storage, transfer, start-up (no JIT compiler) and, likely, memory footpring. 

Presently, the AOT images are *nightly* images, which doesn't instill much confidence with customers looking for a stable version. Please consider providing periodic AOT images given all the benefits they provide.

Thank you!

**Comment by lbussell** [comment]
Hi @simonkurtz-MSFT, thanks for the feedback!

If you are looking for a short-term solution for publishing Native AOT apps, I'd encourage you to look at the [releasesapi Debian sample](https://github.com/dotnet/dotnet-docker/blob/26ac86660467e64de0cf7890fd688374a5b9a0c0/samples/releasesapi/Dockerfile.debian).

In that Dockerfile, you can replace the SDK tag with the Jammy SDK and the runtime deps tag with the Jammy Chiseled runtime deps image to get almost all of the size reduction benefits. We cover this in the [.NET sample image size report](https://github.com/dotnet/dotnet-docker/blob/main/documentation/sample-image-size-report.md#self-contained--trimming-deployment). The only difference between the runtime-deps and the runtime-deps AOT image is the presence of the `libstdc++` library, which ends up being about a 1 MB (compressed) image size savings.

**Comment by lbussell** [comment]
@zeroskyx we currently plan to provide some form of officially-supported Native AOT images for .NET 10. They may not be exactly the same as the AOT images that we currently have in nightly, but there will an easy way to build and host Native AOT apps without installing extra packages for most use-cases. We are still working out what that will look like. You can watch this issue for the latest info.

**Comment by lbussell** [comment]
Here's a new proposal for AOT images. 

1. Create AOT SDK images. These images won't support cross-platform AOT compilation (previously, only the Ubuntu image supported cross-compilation, and it came at an image size and complexity cost). 

3. Change nothing about the OS/Distro offerings for the AOT SDK images, but provide an `aot` convenience tag which points to Ubuntu. I am intentionally choosing to leave Debian out of the AOT image offering because we have not seen requests for it.

5. No AOT-specific runtime-deps images. The regular runtime-deps images, while they contain an extra library or two, are only ~10% larger than their AOT counterparts. At single-digit MB sizes, these are diminishing returns. For reference:    
      - `mcr.microsoft.com/dotnet/nightly/runtime-deps:9.0-alpine3.20-amd64` - 6.61 MB compressed
      - `mcr.microsoft.com/dotnet/nightly/runtime-deps:9.0-alpine3.20-aot-amd64` - 5.70 MB compressed

So, the proposed image offering would look like this:
 
- `dotnet/sdk`
  - `10.0-preview-aot`, `10.0-preview-noble-aot`
  - `10.0-preview-alpine-aot`
  - `10.0-preview-azurelinux3.0-aot`
- `dotnet/runtime-deps`
  - No new images.

/cc @MichaelSimons, @richlander

**Comment by simonkurtz-MSFT** [comment]
@lbussell, I didn't know that the runtime deps and runtime deps aot images are so close. I totally agree with you that shaving off a Megabyte is a micro-optimization that doesn't provide sufficient ROI and, in addition to more work for you all, would cause confusion for users. 

I would be interested to see what the AOT SDK image advantages turn out to be in the flesh compared to status quo. Very happy to take something for a spin with my work from a customer perspective, if it helps you at all.

**Comment by MichaelSimons** [comment]
> Change nothing about the OS/Distro offerings for the AOT SDK images, but provide an aot convenience tag which points to Ubuntu. I am intentionally choosing to leave Debian out of the AOT image offering because we have not seen requests for it.

I think we should be offering a Debian image as well.  I believe our Linux distro matrix should be complete as possible across the image variants.  If we don't do this, then the value is diminished for the distro.

The rest of the proposal LGTM.

**Comment by mthalman** [comment]
> Here's a new proposal for AOT images.

I echo the [above comment](https://github.com/dotnet/dotnet-docker/issues/5020#issuecomment-2660321053) about the distro support for this. Note that this is called out as a [guiding principle](https://github.com/dotnet/dotnet-docker/blob/main/documentation/guiding-principles.md) (see 4.ii)

**Comment by lbussell** [comment]
Good catch on that documentation. I think we can support Debian then. Then the new images would be:

- `dotnet/sdk`
  - `10.0-aot`, `10.0-noble-aot`
  - `10.0-alpine-aot`
  - `10.0-azurelinux3.0-aot`
  - `10.0-trixie-aot`

**Comment by lbussell** [comment]
These images are available in the nightly repos now. They'll be released in the non-nightly repos with .NET 10 Preview 2.

**Comment by simonkurtz-MSFT** [comment]
That's truly great news! Thank you, @lbussell and team! I'll keep the issue open until .NET 10 Preview 2 unless you prefer closing it. If so, please go ahead.

**Comment by lbussell** [comment]
I'll close once it's published, thanks.

**Comment by richlander** [comment]
https://github.com/dotnet/dotnet-docker/discussions/6312

**Comment by simonkurtz-MSFT** [comment]
Most excellent! Thank you very much!
